### PR TITLE
Infra 2313 workflow api enhancements

### PR DESCRIPTION
**Pull request by Sayan-**
JIRA: https://clever.atlassian.net/browse/INFRA-2313

These changes include the createTime enhancement to the job endpoints and the list all workflows from memory store.

- [ ] Update swagger.yml version
- [ ] Run "make generate"
- [ ] After merging, add a github tag to your merge commit

### CODE REVIEW

File: store/dynamodb/dynamodb_store.go [hunk 1]
```diff
@@ -147,6 +147,11 @@ func (d DynamoDB) UpdateWorkflow(def resources.WorkflowDefinition) (resources.Wo
 	return d.GetWorkflow(newVersion.Name(), newVersion.Version())
 }
 
+// TODO
+func (d DynamoDB) GetWorkflows() ([]resources.WorkflowDefinition, error) {
+	return nil, nil
```
Comment (nathanleiby) [inline store/dynamodb/dynamodb_store.go]: nit: you might return an error that says "Not yet implemented" for now

File: store/memory/memory_store.go [hunk 2]
```diff
@@ -50,6 +50,21 @@ func (s MemoryStore) UpdateWorkflow(def resources.WorkflowDefinition) (resources
 	return newVersion, nil
 }
 
+// as a prototype for the moment we will just get the latest version of a given Workflow
+func (s MemoryStore) GetWorkflows() ([]resources.WorkflowDefinition, error) {
```
Comment (mohit) [inline store/memory/memory_store.go]: You might want to add tests for this in the `store/test` package

File: store/memory/memory_store.go [hunk 3]
```diff
@@ -50,6 +50,21 @@ func (s MemoryStore) UpdateWorkflow(def resources.WorkflowDefinition) (resources
...
+	workflows := []resources.WorkflowDefinition{}
+	for k, v := range s.workflows {
+		fmt.Println("k:", k, "v:", v)
```
Comment (mohit) [inline store/memory/memory_store.go]: debug print still here :)

File: store/memory/memory_store.go [hunk 4]
```diff
@@ -50,6 +50,21 @@ func (s MemoryStore) UpdateWorkflow(def resources.WorkflowDefinition) (resources
 	return newVersion, nil
 }
 
+// as a prototype for the moment we will just get the latest version of a given Workflow
```
Comment (mohit) [inline store/memory/memory_store.go]: Comments on public methods in golang need to start with their name:

```golang
// GetWorkflows returns the latest version of all stored Workflows
func (s MemoryStore) GetWorkflows() ([]resources.WorkflowDefinition, error) {
    ...
}
```

File: swagger.yml [hunk 5]
```diff
@@ -33,6 +33,18 @@ paths:
           description: OK response
 
   /workflows:
+    get:
+      operationId: getWorkflows
+      description: Get an array of all available Workflows
```
Comment (mohit) [inline swagger.yml]: we could just have it return the lastest version of all available workflows. And then add a `GET /workflows/{name}/versions` to get a list of all versions available. What do you think?

File: store/memory/memory_store.go (hunk 3 above)
Comment (Sayan-) [inline store/memory/memory_store.go]: Whups, was sure I removed it!

File: swagger.yml (hunk 5 above)
Comment (Sayan-) [inline swagger.yml]: Ah, this sounds like a good way to split the endpoints!

Comment (Sayan-) [inline swagger.yml]: BTW I'll add this in the next PR or make a new ticket for it!

**Comment by mohit** [comment]
This looks good. Before merging we should make sure ark will not have issues with the API change.

**Pull request merged**